src/App.tsx

```diff
@@ -1,61 +1,134 @@
 import './App.scss';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { User } from './types/User';
+import { Todo } from './types/Todo';
+import { TodoList } from './components/TodoList';
+import { FormEvent, useState } from 'react';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+function getUser(userId: number): User | null {
+  const foundUser = usersFromServer.find(user => user.id === userId);
+
+  // if there is no user with a given userId
+  return foundUser || null;
+}
 
 export const App = () => {
+  const [titleTodo, setTitleTodo] = useState('');
+  const [selectedUser, setSelectedUser] = useState('0');
+  const [todos, setTodos] = useState<Todo[]>(
+    todosFromServer.map(todo => ({
+      ...todo,
+      user: getUser(todo.userId),
+    })),
+  );
+  const [errorTitleInput, setErrorTitleInput] = useState('');
+  const [errorUserSelect, setErrorUserSelect] = useState('');
+
+  const handleSubmit = (event: FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    if (!titleTodo && selectedUser === '0') {
+      setErrorTitleInput('Please enter a title');
+      setErrorUserSelect('Please choose a user');
+
+      return;
+    } else if (!titleTodo) {
+      setErrorTitleInput('Please enter a title');
+      setErrorUserSelect('');
+
+      return;
+    } else if (selectedUser === '0') {
+      setErrorTitleInput('');
+      setErrorUserSelect('Please choose a user');
+
+      return;
+    } else {
+      setErrorTitleInput('');
+      setErrorUserSelect('');
+    }
```

Comment:
if you are using returns in if/else, you can just use if without else

Reply:
also you dont need to describe this all errors, you just need to check if title and user are not empty